src/runtime_src/core/include/ert.h

```diff
@@ -172,6 +172,24 @@ struct ert_dpu_data {
   uint32_t chained;                  /* number of following ert_dpu_data elements */
 };
 
+/**
+ * struct ert_cmd_chain_data - interpretation of data payload for ERT_CMD_CHAIN
+ *
+ * @command_count: number of commands in chain
+ * @submit_index:  index of last successfully submitted command in chain
+ * @error_index:   index of failing command if cmd status is not completed
+ * @data[]:        address of each command in chain
+ *
+ * This is the payload of an *ert_packet* when the opcode is ERT_CMD_CHAIN
+ */
+struct ert_cmd_chain_data {
```

Comment:
Can we also have the error status defined in this data structure so that driver does not need to update the error status for each chained command, when the list is processed by device.

Reply:
That's already there is the ert_packet itself.  It is indeed the goal that KMD will no longer update status of individual commands.